pymagicc/io.py

```diff
@@ -1008,6 +1013,50 @@ def _read_magicc6_style_header(self, stream, metadata):
         return column_headers, metadata
 
 
+class _MAGReader(_Reader):
+    def process_header(self, header):
+        """
+        Parse the header for additional metadata.
+
+        Parameters
+        ----------
+        header : str
+            All the lines in the header.
+
+        Returns
+        -------
+        dict
+            The metadata in the header.
+        """
+        metadata = {}
+        in_header = False
+        header_lines = []
+        for line in header.split("\n"):
+            line = line.strip()
+            if not line:
+                continue
+            if line == "---- HEADER ----":
+                in_header = True
+            elif line == "---- METADATA ----":
+                in_header = False
+            else:
+                if in_header:
+                    header_lines.append(line)
+                else:
+                    try:
+                        key, value = line.split(":")
+                    except ValueError:
+                        raise ValueError(  # pragma: no cover # emergency valve
```

Comment:
What's the reason for raising the excepted error?

Reply:
Make the error message clearer

Reply:
Sorry @znicholls , what i meant was why not just print the error message:

```python
line = "no colon"
try:
    key, value = line.split(":")
except ValueError:
    print(  # pragma: no cover # emergency valve
      "Something wrong, line should be of the form 'key: value' "
      "with no other colons (':').\n"
      "Problem line: {}".format(line)
)
```

Output
```
Something wrong, line should be of the form 'key: value' with no other colons (':').
Problem line: no colon
```

Vs.
```python
line = "no colon"
try:
    key, value = line.split(":")
except ValueError:
    raise ValueError(  # pragma: no cover # emergency valve
      "Something wrong, line should be of the form 'key: value' "
      "with no other colons (':').\n"
      "Problem line: {}".format(line)
)
```

Output with both exceptions:
```
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    key, value = line.split(":")
ValueError: not enough values to unpack (expected 2, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 8, in <module>
    "Problem line: {}".format(line)
ValueError: Something wrong, line should be of the form 'key: value' with no other colons (':').
Problem line: no colon
```

Reply:
It doesn’t raise in that case does it? That could be better, I hadn’t actually thought it through

Reply:
I think I changed it in #252 anyway so suggestions welcome there!